### PR TITLE
New Script: Jenkins

### PR DESCRIPTION
**Pull request by quantumryuu**
> [!NOTE]
> We are meticulous when it comes to merging code into the main branch, so please understand that we may reject pull requests that do not meet the project's standards. It's never personal. Also, game-related scripts have a lower chance of being merged.

## Description
Made install script for [Jenkins](https://www.jenkins.io/)

Fixes # (issue)
https://github.com/community-scripts/ProxmoxVE/discussions/400
[Project Board](https://github.com/orgs/community-scripts/projects/9?pane=issue&itemId=88864100)

## Type of change
Please check the relevant option(s):

- [ ] Bug fix (non-breaking change that resolves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (a fix or feature that would cause existing functionality to change unexpectedly)
- [x] New script (a fully functional and thoroughly tested script or set of scripts.)

## Prerequisites
The following efforts must be made for the PR to be considered. Please check when completed:
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Testing performed (I have tested my changes, ensuring everything works as expected)
- [x] Documentation updated (I have updated any relevant documentation)

## Additional Information (optional)
Provide any additional context or screenshots about the feature or fix here.

## Related Pull Requests / Discussions

If there are other pull requests or discussions related to this change, please link them here:
- Related PR #

### CODE REVIEW

**Comment by quantumryuu** [comment]
I removed the update sections and the ${RELEASE} > /opt/${APP}_version because it can be updated with a simple apt update && apt upgrade!

File: ct/jenkins.sh [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  VLAN=""
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+function update_script() {
+header_info
+check_container_storage
+check_container_resources
+}
```
Comment (MickLesk) [inline ct/jenkins.sh:58]: All Missing? No Update? Then Minimum a msg_ like in Adguard-Home

File: install/jenkins-install.sh [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+network_check
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y \
+    curl \
+    mc \
+    sudo \
+    fontconfig \
+    openjdk-17-jre
```
Comment (MickLesk) [inline install/jenkins-install.sh:22]: Is this a available freeware in debian12? Or needed an gpg or something else?

File: install/jenkins-install.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+msg_ok "Installed Dependencies"
+
+
```
Comment (MickLesk) [inline install/jenkins-install.sh:25]: Remove blank line

File: install/jenkins-install.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+msg_info "Installing Jenkins"
```
Comment (MickLesk) [inline install/jenkins-install.sh:26]: Setup Jenkins

File: install/jenkins-install.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+wget -qO /usr/share/keyrings/jenkins-keyring.asc https://pkg.jenkins.io/debian/jenkins.io-2023.key
+echo "deb [signed-by=/usr/share/keyrings/jenkins-keyring.asc]" https://pkg.jenkins.io/debian binary/ >/etc/apt/sources.list.d/jenkins.list
+$STD apt-get update
+$STD apt-get install -y \
+    jenkins
```
Comment (MickLesk) [inline install/jenkins-install.sh:31]: Do an oneliner

File: install/jenkins-install.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+msg_ok "Installed Jenkins"
```
Comment (MickLesk) [inline install/jenkins-install.sh:32]: Setup

File: install/jenkins-install.sh [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
```
Comment (MickLesk) [inline install/jenkins-install.sh:33]: No Service?

**Comment by MickLesk** [comment]
> I removed the update sections and the ${RELEASE} > /opt/${APP}_version because it can be updated with a simple apt update && apt upgrade!

Okay, then do it Like nextpvr or adguardhome :smile:

File: ct/jenkins.sh [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+  DISABLEIP6="no"
+  MTU=""
+  SD=""
+  NS=""
+  MAC=""
+  VLAN=""
+  SSH="no"
+  VERB="no"
+  echo_default
+}
```
Comment (MickLesk) [inline ct/jenkins.sh:53]: can you migrate to new header? Example:

```bash
#!/usr/bin/env bash
source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/misc/build.func)
# Copyright (c) 2021-2024 tteck
# Author: MickLesk (Canbiz) & vhsdream
# License: MIT | https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE
# Source: https://hoarder.app/

# App Default Values
APP="Hoarder"
var_tags="bookmark"
var_cpu="2"
var_ram="4096"
var_disk="8"
var_os="debian"
var_version="12"
var_unprivileged="1"

# App Output & Base Settings
header_info "$APP"
base_settings

# Core
variables
color
catch_errors
```

File: ct/jenkins.sh [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+check_container_resources
+}
+
+start
+build_container
+description
+
+msg_ok "Completed Successfully!\n"
+echo -e "${APP} should be reachable by going to the following URL.
+         ${BL}http://${IP}:8080${CL} \n"
```
Comment (MickLesk) [inline ct/jenkins.sh:66]: can you migrate to new footer? Example:
```bash
msg_ok "Completed Successfully!\n"
echo -e "${CREATING}${GN}${APP} setup has been successfully initialized!${CL}"
echo -e "${INFO}${YW} Access it using the following URL:${CL}"
echo -e "${TAB}${GATEWAY}${BGN}http://${IP}:3000${CL}"
```

**Pull request closed without merging**